apps/prairielearn/src/middlewares/authzHasCoursePreviewOrInstanceView.tsx

```diff
@@ -0,0 +1,53 @@
+import { type Request, type Response } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+
+import { PageLayout } from '../components/PageLayout.js';
+import { getPageContext } from '../lib/client/page-context.js';
+import { Hydrate } from '../lib/preact.js';
+
+import { AuthzAccessMismatch } from './AuthzAccessMismatch.js';
+
+export async function authzHasCoursePreviewOrInstanceView(req: Request, res: Response) {
```

Comment:
This one was a little strange, since it was used by `authzWorkspace`. I returned the html if there was any, and in `authzWorkspace` I send the HTML and return if there is any.

Reply:
> Maybe there's some convenient way to refactor authzHelper.tsx to better support that?
 
 If you have any great ways to do that, feel free!

Reply:
Added 
> I returned the html if there was any, and in authzWorkspace I send the HTML and return if there is any.
back in [d197435](https://github.com/PrairieLearn/PrairieLearn/pull/12496/commits/d19743526ec97d3946689cc82acf37ffa1a32591).